src/applications/vaos/new-appointment/redux/actions.js

```diff
@@ -252,6 +258,74 @@ export function fetchFacilityDetails(facilityId) {
   };
 }
 
+export function openFacilityPageV2(page, uiSchema, schema) {
+  return async (dispatch, getState) => {
+    try {
+      const initialState = getState();
+      const newAppointment = initialState.newAppointment;
+      const typeOfCareId = getTypeOfCare(newAppointment.data)?.id;
+      const userSiteIds = selectSystemIds(initialState);
+      const useVSP = vaosVSPAppointmentNew(initialState);
+      let parentFacilities = newAppointment.parentFacilities;
+
+      dispatch({
+        type: FORM_PAGE_FACILITY_V2_OPEN,
+      });
+
+      if (!parentFacilities) {
+        parentFacilities = await getOrganizations({
+          siteIds: userSiteIds,
+          useVSP,
+        });
+      }
+
+      const responses = await Promise.all(
+        parentFacilities.map(parent => {
+          const parentId = parent.id;
+          const siteId = parseFakeFHIRId(
+            getIdOfRootOrganization(parentFacilities, parentId),
+          );
+          return getSupportedHealthcareServicesAndLocations({
+            siteId,
+            parentId,
+            typeOfCareId,
+            useVSP,
+          });
+        }),
+      );
+
+      const facilities = responses
+        .reduce(function(arr, row) {
+          return arr.concat(row.locations);
+        }, [])
+        .sort((a, b) => {
+          return a.name < b.name ? -1 : 1;
+        });
+
+      dispatch({
+        type: FORM_PAGE_FACILITY_V2_OPEN_SUCCEEDED,
+        page,
+        schema,
+        uiSchema,
+        parentFacilities,
+        facilities,
+        typeOfCareId,
+      });
+    } catch (e) {
+      captureError(e, false, 'facility page');
+      dispatch({
+        type: FORM_PAGE_FACILITY_OPEN_FAILED,
```

Comment:
Should this be the V2 action name? Or should we remove that new const above?

Reply:
Updated to V2 action.  The corresponding reducer doesn't do anything for the scope of this PR but will be updated in PR #3